### PR TITLE
Update chat with account manager banner

### PR DESCRIPTION
**Pull request by mkzie2**
### Fixed Issues

$ https://github.com/Expensify/App/issues/51433
PROPOSAL: https://github.com/Expensify/App/issues/51433#issuecomment-2459450163

### Tests

1. Open Concierge chat.
2. Verify that the banner is updated and displayed correctly.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same

### QA Steps

- Requirement: Have a validated domain admin account
- Add someone as the “Account Manager” via Concierge tool
- Have a domain admin log into NewDot and navigate to the Concierge chat
- Verify there is a “Chat now” button
- Tap the button and verify they are brought to the chat with the Account Manager

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/8f1ca58a-d36a-4d18-bef6-07db3a888e3b

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/fde873ee-863f-4e92-88dc-3ae5f5e358de

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/0590fa58-329b-4bcf-94d4-6fe6f001f6a6

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/e5583a6c-dab3-47df-8750-0586f4894f98

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/dc5dfc02-7a0f-4bf1-9a16-6a8a166d52eb

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/e77c3418-5736-4fcd-b23d-c8e87bbb578c

</details>

### CODE REVIEW

**Comment by danielrvidal** [comment]
@allgandalf was asking for help on the translation but their proposal works great for the translation:

en: Need something specific? Chat with your account manager
es: ¿Necesitas algo específico? Habla con tu gerente de cuenta.
en: Chat now
es: Chatear ahora

**Comment by mkzie2** [comment]
@allgandalf This PR is ready.

**Comment by shawnborton** [comment]
I don't think the banner is supposed to have a hover state:
CleanShot 2024-11-19 at 05 20 06@2x

Let's check with the @Expensify/design though, but I think we should remove that (no background color change, don't change the color of the icon either). 

Also from @dannymcclain 's mock, it looks like maybe the gap between the green button and the close icon should be slightly larger (add 4px more). Here is the mock:
image

**Review by allgandalf — COMMENTED** [review]
Please address the feedback

File: src/styles/index.ts [hunk 1]
```diff
@@ -522,10 +522,6 @@ const styles = (theme: ThemeColors) =>
             lineHeight: variables.lineHeightNormal,
```
[the remark this answers is not in the document]
Comment (mkzie2) [reply src/styles/index.ts:522]: New prettier check fails if we have unused styles, so I had to remove two of them.

**Comment by mkzie2** [comment]
@allgandalf Regarding https://github.com/Expensify/App/pull/52682#issuecomment-2485329178, I mocked `accountManagerReportID` to be self DM report ID, so in the screenshots the button navigates to self DM.

**Comment by mkzie2** [comment]
@allgandalf I updated.

> Please read the expected behaviour below:

I only mocked `accountManagerReportID` to be self DM, not changing any logics in open account manager report. In the case account has an actual AM, it should work correctly.

**Comment by mkzie2** [comment]
@marcaaron How can we create an account with an account manager cc @allgandalf

**Comment by marcaaron** [comment]
I'm confused by the question. Can we not use the trick that I mentioned already and set any `reportID` manually in the Onyx data?

**Comment by marcaaron** [comment]
@allgandalf @mkzie2 let's pull out the stops here? 

Having this show up is not so easy because you need a validated private domain and we would also need to add an account manager for you. We likely have test accounts for internal use. @danielrvidal might be able to set something up for us, but I'd recommend just hardcoding something for test purposes and then we can send this to QA for further testing.

**Comment by mkzie2** [comment]
> @mkzie2 can you please share a reproducible steps video on the ad hoc build showing how to bring the banner up and the redirection to chat please

@marcaaron C+ is requesting me for testing on the ad hoc build why I'm asking about this question. About the dev env, yeah we can test by hardcoding.

**Comment by danielrvidal** [comment]
@allgandalf are you all good to keep making progress or do you need anything from me?

File: src/components/Banner.tsx [hunk 2]
```diff
@@ -68,7 +77,7 @@ function Banner({
     return (
         <Hoverable>
             {(isHovered) => {
-                const isClickable = onClose ?? onPress;
+                const isClickable = onClose && onPress;
```
[the remark this answers is not in the document]
Comment (mkzie2) [reply src/components/Banner.tsx:80]: This is used to prevent the hover [here](https://github.com/Expensify/App/pull/52682#issuecomment-2485281604).

**Review by allgandalf — APPROVED** [review]
Design wise LGTM, lets internal QA this and check the functionality 🚀

File: src/languages/en.ts [hunk 3]
```diff
@@ -39,6 +39,7 @@ import type {
     ChangeTypeParams,
     CharacterLengthLimitParams,
     CharacterLimitParams,
+    ChatWithAMParams,
```
Comment (marcaaron) [inline src/languages/en.ts:42]: I would prefer if this was `ChatWithAccountManagerParams` and everywhere else we do `am` changed to `accountManager`

**Review by marcaaron — CHANGES_REQUESTED** [review]
LGTM but let's fix all the abbreviations and change "am" to "accountManager" wherever it appears.

File: src/languages/en.ts (hunk 3 above)
Comment (mkzie2) [reply src/languages/en.ts]: @marcaaron I updated.

**Review by marcaaron — APPROVED** [review]
Amazing, thanks for the changes!

**Comment by kavimuru** [comment]
@mountiny is this internal QA https://github.com/Expensify/App/pull/52682#issuecomment-2500303934?

**Comment by mountiny** [comment]
Seems like the internal QA is quite tricky, the changes are quite simple so I feel like we do not have to hold deploy on this in case the QA is not by the time to deploy

**Pull request merged**